### PR TITLE
Prepare update to Rust SDK 0.1.36

### PR DESCRIPTION
**Pull request by jmartinesp**
This PR upgrades to the Rust SDK v0.1.36.

Additional changes:

- Adapt `NotificationData` to work without the underlying event for invites.
- Cancel `fetchMembers` when exiting the room.
- Add sonatype maven repo to have immediate access to Rust SDK versions when they're published (no need to wait for maven central synchronisation).

### CODE REVIEW

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/RustMatrixTimeline.kt [hunk 1]
```diff
@@ -100,6 +101,8 @@ class RustMatrixTimeline(
 
     init {
         Timber.d("Initialize timeline for room ${matrixRoom.roomId}")
+
+        val rustTaskHandleBag = TaskHandleBag()
```
Comment (ganfra) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/timeline/RustMatrixTimeline.kt:105]: just call it `taskHandleBag`?

**Review by ganfra — APPROVED** [review]
One small remark otherwise LGTM!

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/notification/NotificationMapper.kt [hunk 2]
```diff
@@ -19,29 +19,60 @@ package io.element.android.libraries.matrix.impl.notification
...
+class NotificationContentMapper(
+    private val sessionId: SessionId,
+) {
+    private val timelineEventToNotificationContentMapper = TimelineEventToNotificationContentMapper()
+
+    fun map(notificationEvent: NotificationEvent): NotificationContent =
+        when (notificationEvent) {
+            is NotificationEvent.Timeline -> timelineEventToNotificationContentMapper.map(notificationEvent.event)
+            is NotificationEvent.Invite -> NotificationContent.StateEvent.RoomMemberContent(
+                userId = sessionId.value,
```
Comment (bmarty) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/notification/NotificationMapper.kt:70]: I am wondering why `RoomMemberContent::userId` is a String and not a UserId, but not blocking here.

**Review by bmarty — APPROVED** [review]
LGTM, nothing blocking. Thanks!

**Pull request merged**